blog: Vibe Coding (#82)
eclipse-che/blog: _posts/2025-06-23-vibe-coding.adoc

---
title: "Vibe Coding"
layout: post
author: Ilya Buziuk
description: >-
  Vibe Coding with Eclipse Che
categories: []
keywords: ['Vibe Coding', 'Roo', 'Eclipse Che', 'Kubernetes']
slug: /@ilya.buziuk/vibe-coding
---
== Introduction

> Imagine a world where coding isn't about meticulous syntax or endless debugging, but about intuition, conversation, and pure flow. Welcome to "Vibe Coding" – a revolutionary approach to software development where Large Language Models (LLMs) become your ultimate co-pilot, transforming complex tasks into effortless interactions.

Author: ```suggestion
Imagine a world where coding isn't about meticulous syntax or endless debugging, but about conversation and pure flow. Welcome to "Vibe Coding" – a revolutionary approach to software development where Large Language Models (LLMs) become your ultimate co-pilot, transforming complex tasks into effortless interactions.
```

== Open-Source Tools for the Vibe Coder

To truly embrace Vibe Coding, you need the right tools. While many closed-source assistants exist—such as link:https://cursor.sh/[Cursor], link:https://www.windsurf.ai/[Windsurf], and link:https://kiro.dev/[Kiro] (all effectively forks of the link:https://github.com/microsoft/vscode[Visual Studio Code - Open Source ("Code - OSS")]), the open-source community offers powerful alternatives that provide greater control and transparency.

IMPORTANT: A significant drawback of local-first AI tools is their potential to disrupt your local development environment if the AI assistant misbehaves. Cloud Development Environments (CDEs), however, offer a crucial advantage for AI assistants: they provide isolated, repeatable, and on-demand workspaces. This means developers don’t have to worry about misbehaving AI tools, MCP agents, or other integrations breaking their local setup, as any issues remain contained within a pod/namespace itself (assuming proper RBAC setup for the developer persona on the Kubernetes cluster).

> Let's explore a few open-source tools available on link:https://open-vsx.org/[Open VSX] that make vibe coding a reality today.

Author: ```suggestion
Let's explore a few awesome open-source extensions available on link:https://open-vsx.org/[Open VSX] that make vibe coding a reality today.
```

=== Continue

> link:https://continue.dev/[Continue] is an open-source autopilot for software development, available under the Apache 2.0 license. Continue enables developers to create, share, and use custom AI code assistants with our open-source VS Code and JetBrains extensions and hub of models, rules, prompts, docs, and other building blocks. The link:https://open-vsx.org/extension/Continue/continue[extension] allows you to connect to any LLM, local or hosted, to receive coding assistance directly within the IDE.

Author: ```suggestion
link:https://continue.dev/[Continue] is an open-source autopilot for software development, available under the Apache 2.0 license. Continue enables developers to create, share, and use custom AI code assistants with open-source VS Code and JetBrains extensions, as well as a hub of models, rules, prompts, documentation, and other building blocks. The link:https://open-vsx.org/extension/Continue/continue[extension] allows you to connect to any LLM, local or hosted, to receive coding assistance directly within the IDE.
```

=== Roo Code

link:https://open-vsx.org/extension/RooVeterinaryInc/roo-cline[Roo Code] is yet another open-source extension for Visual Studio Code, available under the Apache 2.0 license. As a link:https://github.com/RooCodeInc/Roo-Code[fork of Cline], Roo Code builds upon its foundation to provide an AI coding assistant that integrates directly into the IDE. It excels at understanding your intentions, offering intelligent code completions, refactoring suggestions, and explaining complex code blocks. Designed to thrive in CDEs, Roo Code provides a secure, reproducible environment to push the boundaries of AI-powered development. With Roo, you can request broad changes, refactor entire functions with a single prompt, and let the AI handle repetitive tasks, freeing you to focus on the creative aspects of coding. Roo Code is an AI-powered autonomous coding agent that lives in your editor. It can:

- Communicate in natural language
- Read and write files directly in your workspace
- Run terminal commands
- Automate browser actions

> -  ntegrate with any OpenAI-compatible or custom API/model

Reviewer: A typo here

Author: ```suggestion
- integrate with any OpenAI-compatible or custom API/model
```
